python/dicom_archive.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python
+
+from datetime import date
+from typing import Any, cast
+import argparse
+import gzip
+import os
+import shutil
+import sys
+import tarfile
+
+from lib.database import Database
+import lib.dicom.dicom_database
+import lib.dicom.dicom_log
+import lib.dicom.summary_make
+import lib.dicom.summary_write
+import lib.dicom.text
+import lib.exitcode
+
+
+def print_error_exit(message: str, code: int):
+    print(f'ERROR: {message}', file=sys.stderr)
+    sys.exit(code)
+
+
+def print_warning(message: str):
+    print(f'WARNING: {message}', file=sys.stderr)
+
+
+def check_create_file(path: str):
+    if os.path.exists(path):
+        if overwrite:
+            print_warning(f'Overwriting \'{path}\'')
+        else:
+            print_error_exit(
+                (
+                    f'File or directory \'{path}\' already exists. '
+                    'Use option \'--overwrite\' to overwrite it.'
+                ),
+                lib.exitcode.TARGET_EXISTS_NO_CLOBBER,
+            )
+
+
+# Modified version of 'lorisgetopt.load_config_file'.
+# We use argparse to parse the command line options in this script,
```

Comment:
I think it would be best to use the `lorisgetopt.py` library to parse arguments since it checks and has a few other loris specific functionalities. Also, it avoids the duplicated code for `load_config_file.py`. It also avoids having to add another dependency.

Reply:
I am not a fan of `lorisgetopt.py` design, but you are probably right that it is best to avoid duplication, I'll do it.